homeassistant/components/hassio/handler.py

```diff
@@ -46,10 +46,14 @@ def api_data(funct):
 
     async def _wrapper(*argv, **kwargs):
         """Wrap function."""
-        data = await funct(*argv, **kwargs)
-        if data["result"] == "ok":
-            return data["data"]
-        raise HassioAPIError(data["message"])
+        while True:
+            data = await funct(*argv, **kwargs)
+            if data["result"] == "ok":
+                return data["data"]
+            if data["message"] == "System is not ready with state: setup":
```

Comment:
This needs to log something as well or it's likely to hide the issue for years

Reply:
ACK